### PR TITLE
cluster/gce: update cloud-provider-gcp image to latest tagged version

### PR DESCRIPTION
**Pull request by andrewsykim**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Update cloud-provider-gcp image used in CI to latest tagged version based on 1.26

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster/gce/manifests/cloud-controller-manager.manifest [hunk 1]
```diff
@@ -23,7 +23,7 @@
 "containers":[
     {
     "name": "cloud-controller-manager",
-    "image": "gcr.io/k8s-staging-cloud-provider-gcp/cloud-controller-manager:v1.25.2-alpha_ae91c1fc0c443c464a4c878ffa2a4544483c6d1f",
+    "image": "gcr.io/k8s-staging-cloud-provider-gcp/cloud-controller-manager:v26.2.4",
```
Comment (andrewsykim) [inline cluster/gce/manifests/cloud-controller-manager.manifest:26]: Unfortunately I still had to push this tag manually as we don't publish this image yet as part of the cloud-provider-gcp release process. cc @jprzychodzen

File: cluster/gce/manifests/cloud-controller-manager.manifest [hunk 2]
```diff
@@ -23,16 +23,14 @@
...
     "resources": {
       "requests": {
         "cpu": "{{cpurequest}}"
       }
     },
     "command": ["/cloud-controller-manager"],
     "args": [
-      "--log-file=/var/log/cloud-controller-manager.log",
```
Comment (andrewsykim) [inline cluster/gce/manifests/cloud-controller-manager.manifest:34]: Removing temporarily to see if CI will go green, opened https://github.com/kubernetes/cloud-provider-gcp/issues/520 to follow-up

File: cluster/gce/manifests/cloud-controller-manager.manifest (hunk 1 above)
Comment (jprzychodzen) [reply cluster/gce/manifests/cloud-controller-manager.manifest:26]: Can you add commands how you push image to https://github.com/kubernetes/cloud-provider-gcp/blob/master/docs/instruction-how-to-bump-repository.md ?

**Comment by aojea** [comment]
sweeet, it works fine now https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/117605/pull-e2e-gce-cloud-provider-disabled/1651183969196052480/artifacts/e2e-259cc44a81-3e477-master/cloud-controller-manager.log

/lgtm
/approve

@andrewsykim as a follow up, if needed, I can see in the logs the route controller is running

**Comment by aojea** [comment]
/assign @dims

we need a cluster level approver

File: cluster/gce/manifests/cloud-controller-manager.manifest (hunk 2 above)
Comment (aojea) [reply cluster/gce/manifests/cloud-controller-manager.manifest:34]: this comment is stale, referenced issue explain the problem, cloud-provider-gcp moved to go-runner

Comment (dims) [reply cluster/gce/manifests/cloud-controller-manager.manifest:34]: thanks!

**Pull request merged**